src/variables/utils/formatVarsOption.ts

```diff
@@ -1,17 +1,37 @@
 import {OPTION_NAME} from 'src/variables/constants/index'
 import {formatExpression} from 'src/variables/utils/formatExpression'
-import {VariableAssignment} from 'src/types/ast'
+import {VariableAssignment, Variable} from 'src/types'
+import {asAssignment} from 'src/variables/selectors'
 
-export const formatVarsOption = (variables: VariableAssignment[]): string => {
+export const formatVarsOption = (
+  variables: VariableAssignment[] | Variable[]
+): string => {
   if (!variables.length) {
     return ''
   }
 
-  const lines = variables.map(v => `${v.id.name}: ${formatExpression(v.init)}`)
+  const lines = getAssignmentVariables(variables).map(
+    v => `${v.id.name}: ${formatExpression(v.init)}`
+  )
 
   const option = `option ${OPTION_NAME} = {
   ${lines.join(',\n  ')}
 }`
 
   return option
 }
+
+const getAssignmentVariables = (
+  variables: Variable[] | VariableAssignment[]
+): VariableAssignment[] => {
+  const assignments = []
+
+  // Looping over instead of map to get around ts interface signature incompatibility error
+  variables.forEach(v =>
+    v.type === 'VariableAssignment'
+      ? assignments.push(v)
+      : assignments.push(asAssignment(v))
+  )
+
+  return assignments.filter(v => !!v)
+}
```

Comment:
Why not just inline this abstraction in the above function so that we don't create an array copy each time?